### ISSUE TITLE
API download fails to include eeg and extended data

### ISSUE THREAD

**Issue by deniztepe** [OpenEnergyPlatform/open-MaStR#277]
Calling the `Mastr.download()` method with API settings, it makes a different if I give the technology parameter as string or as list of strings. It shouldn't matter. Technology parameter given as string, eeg and extended data are not downloaded. This has to have something to do with parameter parsing in some deeper `MastrMirror` methods. 

```
mastr.download(
    method="API",
    technology="biomass",
    api_processes=api_processes,
    api_limit=limit,
    api_chunksize=chunksize,
    api_date=date,
    api_data_types=data_types,
    api_location_types=location_types,
)
```

**Comment by FlorianK13** [comment]
I would suggest that we change the following: Only the upper method from `mastr.download()` accepts technology as string, list, or None. All methods below this, eg all xml download and soap api methods only accept technology as a list of strings.

Then we check the technology type once at the very beginning and convert string-> list and None->list, from there on we trust that type(technology) = list.

**Comment by chrwm** [comment]
I think this is a good solution!

**Comment by deniztepe** [comment]
Implement after refactoring #272 to avoid possible conflicts.

**Comment by deniztepe** [comment]
Closing, because temporally solved with #327 
Clear implementation should be added with the feature #308
